src/graph/graphcontext.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _GRAPHCONTEXT_H
+#define _GRAPHCONTEXT_H
+
+#include <assert.h>
+#include "../redismodule.h"
+#include "graph.h"
+#include "../index/index.h"
+#include "../stores/store.h"
+
+// This would really belong in store.h, but I'm putting it here momentarily for visiility
+typedef struct {
+  LabelStore *store;
+  char *storename;
+  LabelStoreType type;
+} StoreHandle;
+
+typedef struct {
+  RedisModuleCtx *ctx;
+  Graph *g;
+  char *graph_name;
+
+  // We could possibly move Graph's label and relation cap and count variables
+  // here as well
+  char **relation_strings;
+  char **label_strings;
+
+  Index **indices;
+  int index_ctr;
+
+  LabelStore *edge_all_store;
+  LabelStore *node_all_store;
+  StoreHandle **edge_stores;
+  StoreHandle **node_stores;
+} GraphContext;
```

Comment:
I'm all in favor of this object,
I think it would reduce the number of times we access Redis keys, as there would be just one key (the graph_ctx key) would defiantly simplify the way we look up schemas and indices, that said I think the serialisation / persistency logic would be a bit more complicated